### PR TITLE
Add register with strong password feature

### LINKED ISSUE #897: Allow new user only to register with Strong password
**Is your feature request related to a problem? Please describe.**
In the current implementation, password validations happen at a low level. It accepts the password if it has more than 8 characters. It's not a strong validation checkup.

**Describe the solution you'd like**
Allow users only to register with a High-level password which has mix uppercase, lowercase and numbers, and at least 8 length characters.

### PR DESCRIPTION
**Pull request by sahandilshan**
## Description
With the current implementation of the Bassa app, new users can sing up with a low-level security password. But in this PR I'm validating the password at a high level, so the user has to input a password mix with uppercase, lowercase, and numbers.

## Related Issue
Fix #897 

## Motivation and Context
Strong password improves security, When using a low-level password others can easily guess/hack the password.

## Screenshots (In case of UI changes):
Annotation 2020-03-20 35442
Annotation 2020-03-20 235414
Annotation 2020-03-20 235442
Annotation 2020-03-20 2354142

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by kmehant** [comment]
@sahandilshan Thanks for the pull request 🌟.
Basically in your present design, you are firing all the warnings at a time for every change. 
Can you make a change here, you can write separate functions for separate checks and simultaneously fire only the concerned warnings rather firing all as a single string.

**Comment by sahandilshan** [comment]
> @sahandilshan Thanks for the pull request 🌟.
> Basically in your present design, you are firing all the warnings at a time for every change.
> Can you make a change here, you can write separate functions for separate checks and simultaneously fire only the concerned warnings rather firing all as a single string.

Hi @kmehant,
This change has been made by 74e3f1e544f928d7e94284ea0b72394916fbaa75 commit.

**Comment by kmehant** [comment]
Thanks for the updated work

File: ui/src/app/signup/controllers/SignupCtrl.js [hunk 1]
```diff
@@ -46,6 +47,42 @@
...
+      $scope.loginForm.password.$dirty = false;
+      $scope.loginForm.password.$invalid = false;
+    }
+    function passwordInValid() {
+      $scope.loginForm.password.$valid = false;
+      $scope.loginForm.password.$invalid = true;
+      $scope.loginForm.password.$dirty = true;
+    }
+
+    $scope.validatePwd = function () {
```
Comment (deshankoswatte) [inline ui/src/app/signup/controllers/SignupCtrl.js]:
```suggestion
    $scope.validatePwd = () => {
```
I think this will fix the codacy issues.

**Pull request merged**